File: 02-query/src/createTable.js
const AWS = require("aws-sdk");
const DynamoDB = new AWS.DynamoDB({ region: "us-east-1" });

DynamoDB.createTable(
  {
    TableName: "MovieRoles",
    AttributeDefinitions: [
      {
        AttributeName: "Actor",
        AttributeType: "S",
      },
      {
        AttributeName: "Movie",
        AttributeType: "S",
      },
    ],
    KeySchema: [
      { AttributeName: "Actor", KeyType: "HASH" },
      { AttributeName: "Movie", KeyType: "RANGE" },
    ],
    ProvisionedThroughput: {
      ReadCapacityUnits: 5,
      WriteCapacityUnits: 5,
    },
  },
  function (err) {
    if (err) console.log(`Error creating table: ${err}`);
    else console.log("Table created succesfully! 🚀");
  }
);

File: 02-query/python/create_table.py
import boto3

client = boto3.client("dynamodb", region_name="us-east-1")

try:
    client.create_table(
        TableName="MovieRoles",
        AttributeDefinitions=[
            {"AttributeName": "Actor", "AttributeType": "S"},
            {"AttributeName": "Movie", "AttributeType": "S"},
        ],
        KeySchema=[
            {"AttributeName": "Actor", "KeyType": "HASH"},
            {"AttributeName": "Movie", "KeyType": "RANGE"},
        ],
        ProvisionedThroughput={"ReadCapacityUnits": 5, "WriteCapacityUnits": 5},
    )
    print("Table created successfully! 🚀")
except Exception as err:
    print(f"Error creating table: {err}")
